packages/nimbuild-corejs/lib/module-resolver.js

```diff
@@ -86,9 +87,10 @@ function getCoreJSModulesByUserAgent({features, uaString, logger}) {
     // Build target platform by family and major/minor (if non-zero values)
     const versionArray = version.match(/([\d]+)/g);
 
-    const major = versionArray[0];
+    const major = normalizeUnreleased(family, versionArray[0]);
```

Comment:
This function is expected to either return:
- value defined in `versionArray[0]` which is string major version
- latest supported major version of family, if provided version is higher

This guarantees we are working within the defined dataset in browserslist.